### PR TITLE
Fix aws filters to include productCode for CentOS

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

```release-note
Fixed an issue with the image discovery for CentOS on AWS
```

### CODE REVIEW

File: pkg/cloudprovider/provider/aws/provider.go [hunk 1]
```diff
@@ -95,7 +95,8 @@ var (
 		providerconfig.OperatingSystemCentOS: {
 			description: "CentOS Linux 7 x86_64 HVM EBS*",
 			// The AWS marketplace ID from AWS
-			owner: "679593333241",
+			owner:       "679593333241",
```
Comment (thz) [inline pkg/cloudprovider/provider/aws/provider.go]: is it necessary to use the id instead of "aws-marketplace"? the latter would be better readable.

Comment (alvaroaleman) [reply pkg/cloudprovider/provider/aws/provider.go]: Well, for CoreOS and Ubuntu the owner ids are actually something else than the marketplace. We could probably use both for readability? I am not sure if the name is unique

File: pkg/cloudprovider/provider/aws/provider.go [hunk 2]
```diff
@@ -187,8 +189,21 @@ func getDefaultAMIID(client *ec2.EC2, os providerconfig.OperatingSystem, region
 				Name:   aws.String("root-device-type"),
 				Values: aws.StringSlice([]string{"ebs"}),
 			},
+			{
+				Name:   aws.String("architecture"),
+				Values: aws.StringSlice([]string{"x86_64"}),
```
Comment (thz) [inline pkg/cloudprovider/provider/aws/provider.go]: would it make sense to move the arch decision up to the caller and not hardcode it in getDefaultAMIID?

File: pkg/cloudprovider/provider/aws/provider.go (hunk 1 above)
Comment (thz) [reply pkg/cloudprovider/provider/aws/provider.go]: I just want to make sure the owner is immediately understood. and "aws-marketplace" is less arbitrary id.

Comment (thz) [reply pkg/cloudprovider/provider/aws/provider.go]: Oh well - there is the comment above. So my comment is just about cosmetics.

File: pkg/cloudprovider/provider/aws/provider.go (hunk 2 above)
Comment (alvaroaleman) [reply pkg/cloudprovider/provider/aws/provider.go]: The filters are just as hardcoded and right now we support amd64 only

Comment (thz) [reply pkg/cloudprovider/provider/aws/provider.go]: Just figured that there is no arm offering for it in marketplace anyways.

**Pull request merged**